types/vfile/index.d.ts

```diff
@@ -0,0 +1,85 @@
+// Type definitions for vfile 2.2
+// Project: https://github.com/vfile/vfile#readme
+// Definitions by: Junyoung Choi <https://github.com/rokt33r>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+declare namespace vfile {
+    type Contents = string | Buffer;
+
+    interface Point {
+        line: number;
+        column: number;
+    }
+
+    interface Position {
+        start: Point;
+        end: Point;
+    }
+
+    interface Node {
+        type: string;
+        data?: {};
+        position?: Position;
```

Comment:
Should this have things like `children`, `value`, or basically anything else too?

Reply:
Yeah, it should be better to have them.

```
children?: Node[];
value?: string;
```

It should be fine, right?

Reply:
I think so! Nodes are allowed to have any property though. [HAST](https://github.com/syntax-tree/hast#ast) includes some other props actually in use.

Reply:
I see. Let's just keep only `children` and `value` for now.

To implement typings for HAST, we can extends from the interface

Reply:
`vfile` doesn’t care about any other properties. Just that it’s an object with a `position`, and it’ll use that:

https://github.com/vfile/vfile/blob/86d0c4a17460f3788be7994b182ae8a3bc4207b5/index.js#L169-L171

So you could also allow any other properties!

Reply:
I see. The `message` method doesn't seem to care other values except the same properties of `Position`.
So, I think we should delete `Node` interface and exploit generics again.

```
<P extends Position>(reason: string, position?: Point | P, ruleId?: string) => VFileError;
```

Now the type checking works as our intended. :)
<img width="367" alt="screen shot 2017-09-07 at 2 00 31 am" src="https://user-images.githubusercontent.com/5865853/30124532-6b60f6dc-9370-11e7-96d5-1ad23778f518.png">
<img width="369" alt="screen shot 2017-09-07 at 2 00 51 am" src="https://user-images.githubusercontent.com/5865853/30124535-6d1f80c4-9370-11e7-805d-37cf4dfaca45.png">